undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -295,7 +302,16 @@ private void processDeployment(final WarMetaData warMetaData, final DeploymentUn
                 .addDependency(SuspendController.SERVICE_NAME, SuspendController.class, undertowDeploymentInfoService.getSuspendControllerInjectedValue())
                 .addDependencies(additionalDependencies);
         if(securityDomain != null) {
-            infoBuilder.addDependency(SecurityDomainService.SERVICE_NAME.append(securityDomain), SecurityDomainContext.class, undertowDeploymentInfoService.getSecurityDomainContextValue());
+            if (knownSecurityDomain.test(securityDomain)) {
+                infoBuilder.addDependency(
+                        deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT)
+                                .getCapabilityServiceName(buildDynamicCapabilityName(
+                                        ApplicationSecurityDomainDefinition.APPLICATION_SECURITY_DOMAIN_CAPABILITY,
+                                        securityDomain)),
```

Comment:
Here too CapabilityServiceSupport offers a 2 param method that removes the need to use buildDynamicCapabilityName